sample/src/main/java/io/pixsight/sample/scenemanager/SampleInflateOnDemandActivity.kt

```diff
@@ -0,0 +1,98 @@
+package io.pixsight.sample.scenemanager
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import io.pixsight.scenemanager.SceneCreator
+import io.pixsight.scenemanager.SceneListener
+import io.pixsight.scenemanager.SceneManager
+import io.pixsight.scenemanager.animations.SceneAnimations
+import io.pixsight.scenemanager.annotations.BuildScenes
+import io.pixsight.scenemanager.annotations.Scene
+import kotlinx.android.synthetic.main.main.*
+import kotlinx.android.synthetic.main.placeholder.*
+import kotlinx.android.synthetic.main.spinner.*
+
+@BuildScenes(
+    value = [
+        Scene(scene = Scene.MAIN, viewIds = intArrayOf(
+            R.id.activity_no_annotations_sample_main_content,
+            R.id.activity_no_annotations_sample_main_content_another_view
+        )),
+        Scene(scene = Scene.SPINNER, viewIds = intArrayOf(
+            R.id.activity_no_annotations_sample_main_content_another_view,
+            R.id.activity_no_annotations_sample_loader
+
+        )),
+        Scene(scene = Scene.PLACEHOLDER, viewIds = intArrayOf(
+            R.id.activity_no_annotations_sample_placeholder
+        ))
+    ],
+    inflateOnDemand = true
+)
+class SampleInflateOnDemandActivity : AppCompatActivity(), View.OnClickListener {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_inflate_on_demand_sample)
+        SceneManager.create(
+            SceneCreator.with(this)
+                .first(Scene.MAIN)
+                .animation(SceneAnimations.TRANSLATE_X)
+                .listener(object : SceneListener {
```

Comment:
Did you check if all functions of the listener are triggered only once for one scene change?

Reply:
Oui les events hiding et hidden sont unique pour chaque scene. J'ai ajouté un filtre pour envoyer uniquement les events de la new et last scene.